packages/vc-export/src/Presentation.ts

```diff
@@ -0,0 +1,278 @@
+ * @param options Additional optional configuration.
+ * @param options.validFrom A Date or timestamp (in ms since the epoch) indicating the earliest point in time where the presentation becomes valid.
+ * @param options.validUntil A Date or timestamp (in ms since the epoch) indicating the earliest point in time where the presentation becomes valid.
+ * @param options.expiresIn Duration of validity of the presentation in seconds. If omitted, the presentation's validity is unlimited.
+ * @param options.challenge Optional challenge provided by a verifier that can be used to prevent replay attacks.
+ * @param options.verifier Identifier of the verifier to prevent unintended re-use of the presentation.
+ * @returns A signed JWT in compact representation containing a VerifiablePresentation.
+ */
+export function signAsJwt(
+  presentation: VerifiablePresentation,
+  signingKey: {
+    secretKey: Uint8Array
+    keyUri: DidResourceUri
+    type: supportedKeys
+  },
+  options: {
+    challenge?: string
+    verifier?: string
+    validFrom?: Date | number | string
+    validUntil?: Date | number | string
+  } & Partial<JWTOptions> = {}
+): Promise<string> {
+  const { challenge, verifier, validFrom, validUntil } = options
+  // map options such as audience/verifier and time of validity to their VP/VC representations
+  const issuanceDate =
+    typeof validFrom !== 'undefined'
+      ? new Date(validFrom).toISOString()
+      : undefined
+  const expirationDate =
+    typeof validUntil !== 'undefined'
+      ? new Date(validUntil).toISOString()
+      : undefined
+  const optionsApplied = {
+    issuanceDate,
+    expirationDate,
+    verifier,
+    // already existing values on the presentation take precedence
+    ...presentation,
+  }
+  // JWS replaces any existing proof
+  delete optionsApplied.proof
+  // produce (unencoded) payload where keys on the presentation object are mapped to JWT claims
+  const payload = presentationToPayload(optionsApplied)
+  // add challenge claim to JWTs
+  if (challenge) {
+    payload.nonce = challenge
+  }
+  // encode and add JWS
+  return createJWT(payload, signingKey, options)
+}
+
+/**
+ * Verifies a JWT rendering of a VerifiablePresentation.
+ *
+ * @param token The JWT in compact (string) encoding.
+ * @param options Optional configuration.
+ * @param options.verifier Expected audience/verifier. Verification fails if the aud claim in the JWT is not equal to this value.
+ * @param options.challenge Expected challenge. Verification fails if the nonce claim in the JWT is not equal to this value.
+ * @param options.skewTime Allowed tolerance, in seconds, when verifying time of validity to account for clock skew between two machines. Default: 60s.
+ * @returns An object including the `presentation` (without proof) and the decoded JWT `payload` containing all claims.
+ */
+export async function verifySignedAsJwt(
+  token: string,
+  {
+    verifier,
+    challenge,
+    skewTime = 60,
```

Comment:
it’s not TTL that is of issue here, it’s the question of whether or not you fail validation if it’s one second before not-before or one second after expires-at. And if not, how much tolerance you allow for. Clock skew can lead to the token arriving at the verifier “before it was issued” (according to their System time).
There’s a default, I think 300s or something, but the discovery is kinda bad this way.

Reply:
This is what I mean. Expires-at effectively defines TTL. The producer of the token can increase expires-at by the clock skew tolerance they’re willing to accept.

Reply:
Regarding my suggestion for a better naming, I see now that the option comes from `did-jwt`. They learned to skew time, a godlike power  :D 

And yes, since we’re wrapping their call, it makes sense to also include this option.

Reply:
Yeah, I should clarify: did-jwt makes assertions about `nbf` and `iat` being in the past in addition to `exp` being in the future. The former two are what I’m most concerned about because I can see this go wrong a lot. But based on you feedback I’m inclined to disable assertions on `iat` and set the default skew tolerance to 0. This way, if you set the validFrom (->`nbf`) you can put it a few minutes in the past as Tom suggested.

Reply:
Yay, simpler APIs! :+1: